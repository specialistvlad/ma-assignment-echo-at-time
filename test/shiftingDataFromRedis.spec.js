const FakeRedis = require('./utils/FakeRedis');
const EchoQueue = require('./utils/QueueWithMockedVariables');

describe('shiftingDataFromRedis', () => {
  it('check positive flow of quering to redis', async () => {
    const mockRedisResponse = [
      [
        null,
        [
          '82f2bc06d2fe28ca2bf0:Message 1',
          '216427f50da68fcea198:Message 2',
          'f01aefe1bf6e284a742e:Message 3',
          'cfbcd4ab8b4d38b28c4b:Message 4',
          'cf761d0bf423e1e5a37f:Message 4_2',
          '0587cb69d23ce9387595:Message 5'
        ]
      ],
      [null, 6]
    ];

    const storage = new FakeRedis({ execReturns: mockRedisResponse });
    const echoQueue = new EchoQueue({ storage });
    const result = await echoQueue.shift();
    expect(result).toMatchSnapshot();
    expect(storage.history).toMatchSnapshot();
  });
});
