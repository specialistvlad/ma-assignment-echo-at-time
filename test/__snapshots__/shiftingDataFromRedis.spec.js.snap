// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shiftingDataFromRedis check positive flow of quering to redis 1`] = `
Array [
  "82f2bc06d2fe28ca2bf0:Message 1",
  "216427f50da68fcea198:Message 2",
  "f01aefe1bf6e284a742e:Message 3",
  "cfbcd4ab8b4d38b28c4b:Message 4",
  "cf761d0bf423e1e5a37f:Message 4_2",
  "0587cb69d23ce9387595:Message 5",
]
`;

exports[`shiftingDataFromRedis check positive flow of quering to redis 2`] = `
Array [
  Object {
    "args": Array [],
    "command": "multi",
  },
  Object {
    "args": Array [
      "queue",
      0,
      3500,
    ],
    "command": "zrangebyscore",
  },
  Object {
    "args": Array [
      "queue",
      0,
      3500,
    ],
    "command": "zremrangebyscore",
  },
  Object {
    "args": Array [],
    "command": "exec",
  },
]
`;
